language-support/ts/daml-types/index.ts

```diff
@@ -204,16 +268,22 @@ export const ContractId = <T>(_t: Serializable<T>): Serializable<ContractId<T>>
 
 /**
  * The counterpart of DAML's `Optional T` type.
+ *
+ * @typeparam T The type of the optionally present value.
```

Comment:
We should give a very brief explanation, that just using `null | T` would not work if you have nested `Optional`s. That's where the seemingly complicated `OptionalInner` type is coming from. See https://docs.daml.com/json-api/lf-value-specification.html#optional for an explanation of the `Optional` encoding.

Reply:
See also the last note of the "Primitive DAML Types" section of https://docs.daml.com/daml2ts/index.html.